adiachenko/src/components/FilmForm/index.tsx

```diff
@@ -0,0 +1,214 @@
+import React, {SyntheticEvent, useState} from "react";
+import Select from 'react-select';
+import { ThemeProvider } from "styled-components";
+import { Button, ButtonTheme } from "../Button";
+import { Form } from "./style";
+
+type initialFilmType = {
+    id?: number,
+    title: string,
+    release_date: string,
+    genres: string[],
+    overview: string,
+    poster_path: string,
+    runtime: number
+}
+
+type FilmFormProps = {
+    film?: initialFilmType,
+    edit?: boolean,
+    closeForm: () => void
+};
+
+const initialFilm: initialFilmType = {
+    title: '',
+    release_date: '',
+    genres: [],
+    overview: '',
+    poster_path: '',
+    runtime: 0
+}
+
+const options = [
+    { value: 'Fantasy', label: 'Fantasy' },
+    { value: 'Adventure', label: 'Adventure' },
+    { value: 'Science Fiction', label: 'Science Fiction' },
+    { value: 'Drama', label: 'Drama' },
+    { value: 'Romance', label: 'Romance' },
+    { value: 'Action', label: 'Action' },
+    { value: 'Comedy', label: 'Comedy' },
+    { value: 'Family', label: 'Family' },
+    { value: 'Animation', label: 'Animation' }
+];
+
+const customStyles = {
+    valueContainer: (provided: any) => ({
+        ...provided,
+        height: '40px',
+        background: 'rgba(79,79,79,0.7)'
+    }),
+    indicatorsContainer: (provided: any) => ({
+        ...provided,
+        background: 'rgba(79,79,79,0.7)'
+    }),
+    menu: (provided: any) => ({
+        ...provided,
+        background: 'rgb(79,79,79)',
+        color: 'white',
+        fontSize: '15px'
+    }),
+    placeholder: (provided: any) => ({
+        ...provided,
+        color: 'white'
+    }),
+    control: (provided: any) => ({
+        ...provided,
+        background: 'rgb(79,79,79)',
+    }),
+};
+
+const FilmForm = ({film, edit, closeForm}: FilmFormProps) => {
+    const [filmSate, setFilm] = useState(film);
+    const resetForm = (event: SyntheticEvent) => {
+        event.preventDefault();
+        if (edit) {
+            setFilm(film);
+        } else {
+            setFilm(initialFilm);
+        }
+    };
+
+    const submitForm = (event: SyntheticEvent, save: boolean) => {
```

Comment:
It looks like you can be more specific with something, like `FormEvent<HTMLFormElement>` in place of `SyntheticEvent`

Reply:
Was decided to use MouseEvent instead FormEvent<HTMLFormElement>
As this event are triggered onClick event on Button
`                {
                    edit ? (
                        <Button
                            onClick={(event) => submitForm(event, true)}
                        >Save</Button>
                    ) : (
                        <Button
                            onClick={(event) => submitForm(event, false)}
                        >Submit</Button>
                    )
                }`